cmake/modules/hunter_add_repository.cmake

```diff
@@ -0,0 +1,81 @@
+include(CMakeParseArguments) # cmake_parse_arguments
+include(hunter_internal_error)
+include(hunter_fatal_error)
+include(hunter_test_string_not_empty)
+include(hunter_status_debug)
+include(hunter_status_print)
+include(hunter_install_repository)
+
+function(hunter_add_repository)
+  if(PROJECT_NAME OR NOT HUNTER_GATE_ROOT)
```

Comment:
I haven't worked with sub-projects in the past, so I'm not sure I understand the pros and cons completely. But I think that's what I want... If I'm not mistaken, this is also the behavior for HunterGate?

In practice, all `hunter_add_repository` must happen before the call to `hunter_finalize` (or more accurately, before the call to `hunter_calculate_config_sha1`) so that the extra packages appear  in the unified `config.cmake`.  Technically, `hunter_add_repository` could be allowed at any point until the first `hunter_add_package`, but forcing inclusion before `project` seems cleaner to me.

Does that sound right?